hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestColumnStatsIndex.scala

```diff
@@ -1056,4 +1060,61 @@ class TestColumnStatsIndex extends ColumnStatIndexTestBase {
       assertTrue(r.getMinValue.asInstanceOf[Comparable[Object]].compareTo(r.getMaxValue.asInstanceOf[Object]) <= 0)
     })
   }
+
+  @Test
+  def testDeserializeFromByteBuffer(): Unit = {
+    // Original decimal value: 123.45 (scale 2)
+    val original: JBigDecimal = new JBigDecimal("123.45")
+    // Get the unscaled value (12345) as a byte array.
+    val schema: Schema = DecimalWrapper.SCHEMA$.getField("value").schema()
+    val buffer: ByteBuffer = ColumnStatsIndexSupport.decConv.toBytes(original, schema, schema.getLogicalType)
+
+    // Create a dummy DecimalWrapper that returns a ByteBuffer.
+    val wrapper: DecimalWrapper = new DecimalWrapper(buffer)
+    // Call tryUnpackValueWrapper – it should match the DecimalWrapper case and return the ByteBuffer.
+    val unwrapped: Any = ColumnStatsIndexSupport.tryUnpackValueWrapper(wrapper)
+    assertTrue(unwrapped.isInstanceOf[ByteBuffer], "Expected a ByteBuffer")
+
+    // Now deserialize the ByteBuffer to a BigDecimal.
+    val dt = DecimalType(10, 2)
+    val deserialized: Any = ColumnStatsIndexSupport.deserialize(unwrapped, dt)
+    assertTrue(deserialized.isInstanceOf[JBigDecimal], "Deserialized value should be a java.math.BigDecimal")
+    assertEquals(original, deserialized.asInstanceOf[JBigDecimal], "Decimal value from ByteBuffer does not match")
+  }
+
+  @Test
+  def testDeserializeFromJavaBigDecimal(): Unit = {
+    // Original decimal value: 543.21
+    val original: JBigDecimal = new JBigDecimal("543.21")
+    // Create an anonymous DecimalWrapper that returns a ByteBuffer of java.math.BigDecimal directly.
+    val wrapper: DecimalWrapper = new DecimalWrapper {
+      val schema: Schema = DecimalWrapper.SCHEMA$.getField("value").schema()
+
+      override def getValue: ByteBuffer = ColumnStatsIndexSupport.decConv.toBytes(original, schema, schema.getLogicalType)
+    }
+
+    val dt = DecimalType(10, 2)
+    val deserialized: Any = ColumnStatsIndexSupport.deserialize(ColumnStatsIndexSupport.tryUnpackValueWrapper(wrapper), dt)
+    assertTrue(deserialized.isInstanceOf[JBigDecimal], "Deserialized value should be a java.math.BigDecimal")
+    assertEquals(original, deserialized.asInstanceOf[JBigDecimal], "Decimal value from java.math.BigDecimal does not match")
+  }
+
+  @Test
+  def testDeserializeFromScalaBigDecimal(): Unit = {
```

Comment:
is it possible to make all 3 into just 1 test and make it parametrized. input value, expected values etc.